content/src/content/jcr_root/apps/core/wcm/components/carousel/v2/carousel/clientlibs/editorhook/js/panelcontainer.js

```diff
@@ -0,0 +1,31 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+(function(channel) {
+    "use strict";
+
+    channel.on("cq-editor-loaded", function(event) {
+        if (window.CQ && window.CQ.CoreComponents && window.CQ.CoreComponents.panelcontainer &&
+            window.CQ.CoreComponents.panelcontainer.v1 && window.CQ.CoreComponents.panelcontainer.v1.registry) {
+            window.CQ.CoreComponents.panelcontainer.v1.registry.register({
+                name: "cmp-carousel",
```

Comment:
given that this is the same as the v1 `cmp-carousel` registration, we can probably remove the editor hook in the v2 Carousel. All editor hooks are loaded into the editor, so we would prevent loading unnecessary code there. 

For future versions that break the selector and item contract we can register them under `name: "cmp-carousel-vX`